woocommerce/woocommerce
# Implementation of separate variant slots via refactored components

### Changes proposed in this Pull Request:

Previously the variant products used the same slots as the parent form, making it impossible to target one and not the other. This refactors the sections and uses discrete IDs to ensure both can be targeted individually. 

I've also reorganized the IDs into a hierarchy of sorts, and other minor optimizations to make things clearer.

Closes #36610 .

### How to test the changes in this Pull Request:

1. Enable new product editor at WooCommerce -> Settings -> Advanced -> Features -> Try the new product editor
2. Enable the `product-variation-management` feature flag with the WC Beta Tester Plugin.
3. Navigate to Product -> Add New to see the new editor.
4. Add the following fills file and import:
```js
/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { registerPlugin } from '@wordpress/plugins';
import {
	__experimentalWooProductTabItem as WooProductTabItem,
	__experimentalWooProductSectionItem as WooProductSectionItem,
	__experimentalProductSectionLayout as ProductSectionLayout,
	__experimentalWooProductFieldItem as WooProductFieldItem,
} from '@woocommerce/components';

const TestFills = () => {
	return (
		<>
			<WooProductTabItem
				id="tab/test"
				templates={ [ { name: 'tab/variation', order: 232 } ] }
				pluginId={ 'woocommerce' }
				tabProps={ { name: 'test_tab', title: 'Test Tab' } }
			>
				<WooProductSectionItem.Slot tab="tab/test" />
			</WooProductTabItem>
			<WooProductSectionItem
				id="tab/test"
				tabs={ [ { name: 'tab/test', order: 1 } ] }
				pluginId={ 'woocommerce' }
			>
				<ProductSectionLayout
					title={ __( 'Test section', 'woocommerce' ) }
					description={ __(
						'Test section description.',
						'woocommerce'
					) }
				>
					<strong>Test fill</strong>
				</ProductSectionLayout>
			</WooProductSectionItem>
			<WooProductFieldItem
				id="field/test"
				sections={ [ { name: 'section/test', order: 1 } ] }
				pluginId={ 'woocommerce' }
			>
				<strong>Test fill</strong>
			</WooProductFieldItem>
			<WooProductFieldItem
				id="field/test"
				sections={ [
					{
						name: 'tab/pricing/basic',
						order: -1,
					},
				] }
				pluginId={ 'woocommerce' }
			>
				<strong>Test field</strong>
			</WooProductFieldItem>
		</>
	);
};

registerPlugin( 'wc-admin-product-editor-form-tab-testfills', {
	// @ts-expect-error 'scope' does exist. @types/wordpress__plugins is outdated.
	scope: 'woocommerce-product-editor',
	render: () => {
		return <TestFills />;
	},
} );

```
5. Notice that the above tab fill titled "Test tab" shows up only on variants.
6. Notice that the "Test field" bold test appears at the top of the pricing section on parent products, but not on variants.
7. Do a general smoke test of the parent and variable product forms.



### Other information:




### FOR PR REVIEWER ONLY:
